sync-core/src/main/java/com/cloudant/sync/query/UnindexedMatcher.java

```diff
@@ -258,7 +265,7 @@ private boolean executeSelectorTree(QueryNode node, DocumentRevision rev) {
     private boolean valueCompare(Object actual, String operator, Object expected) {
         boolean passed;
 
-        if (operator.equals("$eq")) {
+        if (operator.equals("$eq") || operator.equals("$in")) {
```

Comment:
Or, perhaps do the flip of operator to $in in the caller itself where you're doing other pieces of transformation, meaning $in logic is more consolidated in one place.